src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/cas3/Cas3BookingGapReportRepository.kt

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.cas3
+
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import org.springframework.stereotype.Repository
+
+val QUERY = """
+select probation_region,
+  premises_id,
+  premises_name,
+  town,
+  bed_id,
+  bed_name,
+  year,
+  month,
+  days_booked,
+  (DATE_PART('days',
+    DATE_TRUNC('month', make_date(year, month, 1))
+      + '1 MONTH'::INTERVAL
+      - '1 DAY'::INTERVAL) - days_booked)::int as gap_days
+  from (select
+    probation_region,
+    premises_id,
+    premises_name,
+    town,
+    bed_id,
+    bed_name,
+    extract(year from booking_date)::int as year,
+    extract(month from booking_date)::int as month,
+    count(*) as days_booked
+    from (select
+      probation_regions.name as probation_region,
```

Comment:
Returns all "temporary-accommodation" bookings for all regions i.e. no region filter or date filter.